src/VerifiablePresentationManager.ts

```diff
@@ -8,6 +8,36 @@ import {
 } from './Credential';
 import { PresentationVerifier, VerifyFunction } from './PresentationVerifier';
 
+/**
+ * @param {Object} fields: a JSON properties fields object
+ * @param {Object} paths: already calculated paths to be appended to
+ * @param {String} pathPrepend: existing path for nested key in question
+ * @returns {Array}: an array of path objects of type
+ *  {
+ *    {String} path
+ *    {String}: type
+ *    {String}: description
+ *  }
+ */
+const NESTED_PATH_DELIMITER = '.';
+const getFlattenedPaths = (objToMatch: { [prop: string]: any }, paths: any[], pathPrepend: string = '') => {
+    let localPaths: any[] = paths;
+    R.keys(objToMatch).forEach((key: string) => {
```

Comment:
This is a bit confusing to read - in one case the localpaths is being pushed to, but in the other case it is being overwritten. Is this correct? Is there any way we can simplify it?

Reply:
Also consider R.mapObjectIndexed ( I think there is an R.forEachObjectIndexed) to iterate on the key and value out of an object at the same time, avoiding the need for line 26.

Reply:
updated

Reply:
See my follow-up comment below